js/web/app.js

```diff
@@ -0,0 +1,116 @@
+import editor from "./editor.vue"
+
+const routes = [
+    {
+      path: "/:fileId/:filePath/:mode",
+      components: {
+        fullscreen: editor
+      },
+      name: "editor",
+      meta: { hideHeadbar: true }
+    }
+];
+
+const appInfo = {
+    name: "Richdocuments",
+    id: "richdocuments",
+    icon: "x-office-document",
+    isFileEditor: true,
+    extensions: [
+        {
+          extension: "odt",
+          routeName: "richdocuments-editor",
+          newTab: true,
+          newFileMenu: {
+            menuTitle ($gettext) {
+              return $gettext("Document")
+            },
+            icon: "x-office-document"
+          },
+          routes: [
+            "files-personal",
+            "files-favorites",
+            "files-shared-with-others",
+            "files-shared-with-me",
+            "files-public-list"
```

Comment:
Your current implementation doesn't support public links - which is fine for a first iteration. But please remove `files-public-list` from the routes whitelist, so that collabora is not offered as editor on public links for now. Same for the other file extensions.